File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) 
    {
        map<int, int> frq;
        for (const int num : nums)
            frq[num]++;

        vector<pair<int, int>> frqs(frq.begin(), frq.end());

        sort(frqs.begin(), frqs.end(), [](const pair<int, int> &a, const pair<int, int> &b) {
            return a.second > b.second;
        });

        vector<int> ans;

        for (int i = 0; i < k; i++)
            ans.push_back(frqs[i].first);

        return ans;    
    }
};

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.java
class Solution {
    public int findDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<>();

        int ans = 0;

        for(int num: nums) {
            if(set.contains(num)) {
                ans = num;
                break;
            }

            set.add(num);
        }

        return ans;
    }
}
